src/main/java/vazkii/botania/client/integration/jei/JEIBotaniaPlugin.java

```diff
@@ -63,8 +75,16 @@ public void registerCategories(IRecipeCategoryRegistration registry) {
 				new RunicAltarRecipeCategory(registry.getJeiHelpers().getGuiHelper()), // Runic must come before petals. See williewillus/Botania#172
 				new PetalApothecaryRecipeCategory(registry.getJeiHelpers().getGuiHelper()),
 				new ElvenTradeRecipeCategory(registry.getJeiHelpers().getGuiHelper()),
-				new ManaPoolRecipeCategory(registry.getJeiHelpers().getGuiHelper())
-				);
+				new ManaPoolRecipeCategory(registry.getJeiHelpers().getGuiHelper()),
+				new OrechidRecipeCategory(registry.getJeiHelpers().getGuiHelper()),
+				new OrechidIgnemRecipeCategory(registry.getJeiHelpers().getGuiHelper())
+		);
+	}
+
+	public static boolean doesOreExist(Map.Entry<String, Integer> entry) {
```

Comment:
signature is a bit weird for the name, why not just pass the key only?

Reply:
Mostly because I'm using it to filter `oreWeights.entrySet()` and chose to do `filter(JEIBotaniaPlugin::doesOreExist)`. Could just as easily do `filter(e -> doesOreExist(e.getKey()))`